File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        j= len(nums)-1
        
        for i in range(len(nums)-1, -1,-1):
            if i + nums[i] >= j:
                j = i
        
        return True if j==0 else False

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int arr=0;
        for (int i=0 ; i< nums.length ; i++){
            if(arr<i){
                return false;
            }
            arr= Math.max(arr, i+nums[i]);
        }
        return true ;
    }
}